Automattic/wp-calypso
# Design picker: Use variation discs to change the theme thumbnail

Fixes https://github.com/Automattic/dotcom-forge/issues/1658

## Proposed Changes

Changes the behavior of the style variations discs in the onboarding's design picker so they replace the theme thumbnail:

https://user-images.githubusercontent.com/1233880/220145265-771f2299-9c6a-4d81-9c87-95881d2ac904.mov

Some notes about the new behavior:

- The selected style variation is highlighted with a blue border
- The first disc is the default variation (free) for regular themes, and the preselected variation (paid) for virtual themes
- The "Free" / "Premium" badges are dynamically set based on the selected variation
- Clicking on the "+X" button (only visible when there are more than 4 variations) goes to the next step (design preview) with the default variation


## Testing Instructions

- Use the Calypso live link below
- Go to `/setup/site-setup/designSetup?siteSlug=<SITE_DOMAIN>&flags=virtual-themes/onboarding`
- For regular themes, make sure the first disc represents the default variation
- For virtual themes, make sure the first disc represents the preselected variation
- Use the discs to change the selected variation
- Make sure the thumbnails changes accordingly
- Make sure the "Free" / "Premium" badges change accordingly
- Click on the thumbnail
- Make sure the design preview keeps the same style variation selected
- Go back
- Click on the "+X" button on a theme with more than 4 variations
- Make sure the design preview is opened with the default style variation
